views.go

```diff
@@ -73,6 +73,19 @@ func ProcessViewResult(result ViewResult, params map[string]interface{},
 
 	var collator JSONCollator
 
+	if keys, ok := params["keys"].([]interface{}); ok {
+		filteredRows := make(ViewRows, 0)
+		for _, targetKey := range keys {
+			i := sort.Search(len(result.Rows), func(i int) bool {
+				return collator.Collate(result.Rows[i].Key, targetKey) >= 0
+			})
+			if collator.Collate(result.Rows[i].Key, targetKey) == 0 {
```

Comment:
>> If there is no such index, Search returns n. (Note that the "not found" return value is not -1 as in, for instance, strings.Index.)
>
> from: https://golang.org/pkg/sort/#Search

We'll need a length check here to prevent a panic when the `targetKey` is not found in `result.Rows`

```suggestion
			if i < len(result.Rows) && collator.Collate(result.Rows[i].Key, targetKey) == 0 {
```

Reply:
Good catch - fixed.